api-participant/src/main/java/bio/terra/pearl/api/participant/service/CurrentUserService.java

```diff
@@ -38,46 +42,46 @@ public CurrentUserService(
   }
 
   /**
-   * B2C handles all login logistics -- all this method does is confirm the user is in the DB, and
-   * update their last login time
+   * Validate token and verify user represented in token is authorized for portal. If so update user
+   * last login time
    */
   @Transactional
-  public Optional<UserWithEnrollees> tokenLogin(
+  public UserWithEnrollees tokenLogin(
       String token, String portalShortcode, EnvironmentName environmentName) {
-    Optional<UserWithEnrollees> userOpt = loadByToken(token, portalShortcode, environmentName);
-    userOpt.ifPresent(
-        userAndEnrolles -> {
-          userAndEnrolles.user.setLastLogin(Instant.now());
-          participantUserDao.update(userAndEnrolles.user);
-        });
+    UserWithEnrollees userOpt = loadByToken(token, portalShortcode, environmentName);
+    userOpt.user.setLastLogin(Instant.now());
+    participantUserDao.update(userOpt.user);
     return userOpt;
   }
 
   @Transactional
-  public Optional<UserWithEnrollees> refresh(
+  public UserWithEnrollees refresh(
       String token, String portalShortcode, EnvironmentName environmentName) {
     return loadByToken(token, portalShortcode, environmentName);
   }
 
```

Comment:
Nice, I like this refactor to return the object instead of an Optional and handle exception throwing here -- makes the other code more readable